src/rcamera.h

```diff
@@ -134,434 +148,401 @@ void SetCameraMoveControls(int keyFront, int keyBack,
+
+        // Clamp view down
+        float max_angle_down = Vector3Angle(Vector3Negate(up), target_position);
+        max_angle_down *= -1.0f; // downwards angle is negative
+        max_angle_down += 0.001f; // avoid numerical errors
+        if (angle < max_angle_down) angle = max_angle_down;
+    }
+
+    // Rotation axis
+    Vector3 right = GetCameraRight(camera);
+
+    // Rotate view vector around right axis
+    target_position = Vector3RotateAxis(target_position, right, angle);
+
+    if (camera->mode == CAMERA_FREE)
+    {
+        // Move target relative to position
+        camera->target = Vector3Add(camera->position, target_position);
+
+        // Rotate up direction around right axis
+        camera->up = Vector3RotateAxis(camera->up, right, angle);
+    }
+    else if (camera->mode == CAMERA_FIRST_PERSON)
+    {
+        // Move target relative to position
+        camera->target = Vector3Add(camera->position, target_position);
+    }
+    else if (camera->mode == CAMERA_THIRD_PERSON ||
+             camera->mode == CAMERA_ORBITAL)
+    {
+        // Move position relative to target
+        camera->position = Vector3Subtract(camera->target, target_position);
     }
 }
 
-// Set camera pan key to combine with mouse movement (free camera)
-void SetCameraPanControl(int keyPan) { CAMERA.panControl = keyPan; }
+// Rotates the camera around its forward vector
+// Roll is "turning your head sideways to the left or right"
+// Note: angle must be provided in radians
+void CameraRoll(Camera3D *camera, float angle)
+{
+    // Rotation axis
+    Vector3 forward = GetCameraForward(camera);
 
-// Set camera alt key to combine with mouse movement (free camera)
-void SetCameraAltControl(int keyAlt) { CAMERA.altControl = keyAlt; }
+    // Rotate up direction around forward axis
+    camera->up = Vector3RotateAxis(camera->up, forward, angle);
+}
 
-// Set camera smooth zoom key to combine with mouse (free camera)
-void SetCameraSmoothZoomControl(int szoomKey) { CAMERA.smoothZoomControl = szoomKey; }
+// Moves camera slightly to simulate a walking motion
+// Note: Only active if camera->swingCounter > 0
+void CameraViewBobbing(Camera3D* camera)
+{
+    if (camera->swingCounter > 0)
+    {
+        // NOTE: We delay the target movement relative to the position movement to create a little pitch with each step.
+        camera->position.y = camera->position.y - 0.25f * sinf((camera->swingCounter + 1) / CAMERA_FIRST_PERSON_STEP_TRIGONOMETRIC_DIVIDER) / CAMERA_FIRST_PERSON_STEP_DIVIDER;
+        camera->target.y = camera->target.y - 0.25f * sinf((camera->swingCounter - 1) / CAMERA_FIRST_PERSON_STEP_TRIGONOMETRIC_DIVIDER) / CAMERA_FIRST_PERSON_STEP_DIVIDER;
+
+        // Update counter for next frame
+        camera->swingCounter %= 2147483647 /* INT_MAX */; // Counter must be positive
+        camera->swingCounter++;
+    }
+}
+
+// Returns the camera view matrix
+Matrix GetCameraViewMatrix(Camera3D *camera)
```

Comment:
Probably no need for this function. It adds a dependency to `raymath` and  users can already get the matrix with the camera exposed parameters.

Reply:
This function was meant for convenience reasons and for the stand alone mode.
Anyway it can be removed and users just have to call `raymath` themselves.